src/telemetry/telemetryController.ts

```diff
@@ -58,34 +58,44 @@ type NewConnectionTelemetryEventProperties = {
   is_used_saved_connection: boolean;
 };
 
+type DocumentUpdatedTelemetryEventProperties = {
+  source: string;
+  success: boolean;
+};
+
 export type TelemetryEventProperties =
   | PlaygroundTelemetryEventProperties
   | LinkClickedTelemetryEventProperties
   | ExtensionCommandRunTelemetryEventProperties
-  | NewConnectionTelemetryEventProperties;
+  | NewConnectionTelemetryEventProperties
+  | DocumentUpdatedTelemetryEventProperties;
 
 export enum TelemetryEventTypes {
   PLAYGROUND_CODE_EXECUTED = 'Playground Code Executed',
   EXTENSION_LINK_CLICKED = 'Link Clicked',
   EXTENSION_COMMAND_RUN = 'Command Run',
   NEW_CONNECTION = 'New Connection',
   PLAYGROUND_SAVED = 'Playground Saved',
-  PLAYGROUND_LOADED = 'Playground Loaded'
+  PLAYGROUND_LOADED = 'Playground Loaded',
+  DOCUMENT_UPDATED = 'Document Updated'
 }
 
 /**
  * This controller manages telemetry.
  */
 export default class TelemetryController {
```

Comment:
Just curious, why is this called `Controller`? this doesn't strike me as something that receives commands or requests from the UI, but rather something more internally called by other parts of our code, how is the lingo for VSCode? Is this not a controller as in MVC or similar?

Reply:
Do you have a better naming in mind? Maybe telemetryModule? 🤔